### DISCUSSION TITLE
Removal of old tags hinders finding new content

### DISCUSSION THREAD

**Discussion by EratoNysiad** [OpenTTD/OpenTTD#10509] [Features]
Recently a change was pushed to BaNaNaS which added automated tags. I think it's very nice to be able to specifically look for trainsets if I don't know yet what I want and want to just browse, and I certainly don't think that that should be removed.
In the process of adding these new tags, it seems all the old tags have been deleted. Now these tags had in the past been very convenient and useful for finding stuff within a certain theme, even if this specific theme wasn't in the actual name of the NewGRF. For example, OpenTTD has a large variety of Japanese Newgrfs that don't have "Japan" in the title, and not all of them necessarily have it in the description and some of the NewGRF developers are no longer active in the community and would not be able to enact changes in any new system.
Some examples are the JP+ set, TIAS, NATS, Okinawa train set, RIMS and FRIMS. None of these show up when searching for Japan in BaNaNaS.
![image](https://user-images.githubusercontent.com/10280358/221359266-618a7dcc-af53-4e0f-94f5-3d8e6390d700.png)
Many of these do have "Japan" in the description, but can you guarantee that for all NewGRFs that the relevant key words are in the description?

Another problem arises when you get NewGRFs with an abbreviaton for a name. RIMS can now not be found by searching for RIMS, same for TIAS and several others. This wasn't a problem before, as the abbreviation could safely be in the tags.

I think the only real, workable solution to this is to bring back all old tags, which may then be supplemented with the new automated tags, if so desired.

**Comment by TrueBrain** [comment]
Although I fully understand the sentiment "let's go back, it was better there", I rather invest some time and have a conversation about what is actually missing without tags. And as such, I really appreciate posts like this. So if you don't mind, I would like to indulge you a bit more, and see if we can find what we exactly miss without tags.

Honestly, personally, I completely overlooked things like "Japan" were only in tags. Additionally, I was under the impression the search also looked in the description, but that turns out to be a lie :D I can only apologize for that, as I wouldn't have pushed this to production if I knew that was the case.

That said, the tags system as we used to have, was not giving the value we were hoping for. The hope was that over the years the community would figure out a common system to use, and tags would converge. Bit naive most likely, but pretty much the opposite happened :) And this wasn't hurting content writers, but it was our end-user. For example, the amount of different variations we have in tags for "road vehicle" is insane .. so good odds an end-user could find your set, but not a competing set very similar.

So, this is why we have added the classification system. Now for NewGRFs, but soon for the other content-types too. And I am glad to read that the new classification system is liked (at least, by you :D Feel free to let us know if you agree / have another opinion).

Which brings us to your request: bring back the old tags. If you wouldn't mind to hold off on that, I am much more curious what we are exactly missing without them. As there might be nicer / better ways to add them back. And I mean nicer / better here in the context of our end-user. As in the end we all want them to find the right content for them.

From your post I read very clearly: region! And I fully agree. That sounds like an excellent field to add. Of course we can debate how that field should look, but let's put a pin in that, and first see what else we are missing.

The other one I noticed is: authors. Some, in fact, very few, added their own name in the tags, so people could find all their content. Which 100% makes sense to me, but .. I rather have that work for everyone. Not the select few that were smart enough to do this :)

Is there anything else that we are now really missing? Something the end-user would know about and would care about?

The reason I add this last part, is that we also had tags like `asdf` and `testing` and `nightly`. Pretty sure the end-user doesn't actually care about that :)

Floor is yours! Let me know :)

**Reply by EratoNysiad** [reply]
To preface, I'd like to respond to "The hope was that over the years the community would figure out a common system to use, and tags would converge". When I was a brand new NewGRF developer I did see the remark to "try to only use tags other people are using" but I don't remember if there was ever a list of tags in use, though I remember looking for it. But with nothing to go off of, I was forced to ignore the recommendation.

As alread mentioned, region and authors are definitely good.

For authors I would like to raise the point of multi-developer sets. Now I don't have many examples, but JP+ MUs is credited to Tintinfan if you just go by who put it on BaNaNaS, but a lot of sprites were provided by RIPper_KR. He also made some great stuff for NATS, and it may not be immediately obvious how you would lift the proper credits from the newgrfs.

I would say types of trains or road vehicles is also interesting. Narrow gauge, Metro, Monorail, Maglev are all things I sometimes see people asking about. Buses, trucks, trams and eyecandy road vehicle sets all exist side by side and I frequently find myself looking for a tram or truck set specifically.

Some NewGRFs make use of specific features that players may be interested in, which is, for example NRT, and sometimes even features that only exist in JGR, like NRS. U&RaTT and the American Road Replacement Set are both certainly roadsets, but in the NRT era I would much prefer to see U&RaTT. Similarly the "Trolleybus Set" adds trams that pretend to be buses, because that's how that used to work. STGFX Trolleybus set and the North Korean Road Vehicles set 2.0 specifically do add NRT-compatible trolleybuses.
The "Road Stops Extension" is currently listed as "unknown" which is absolutely fair, but I do think this is the first of many.

There may be some others but this is all I can come up with from the top of my head.

**Reply by TrueBrain** [reply]
Cheers, I love this insight :) Tnx a lot!

Let me write out some things that came to mind when reading this reply, mostly so I remember myself when I read this back later.

Authors:
The backend supports multiple developers; currently it also means they can both upload new content for that newgrf-id, but I assume that is not actually a problem. Just currently it isn't possible to add developers to the list in the frontend, but that is a solvable problem :)

Types of trains/roadveh:
I have been wondering about this. My main issue with that is that many sets introduce new types, and that is not always clear to the end-user. So ideally, I would also like to see that as classifier, so we can extend the in-game GUI with a dropdown list much like the website now has for other things. That way it becomes much more clear to the end-user what content is offered, and how many things we have to offer. This is now often lost to the end-user that doesn't really know what to look for. 

NRT:
"Is this something an end-user will understand?" I am wondering. NRT for sure means little to them (them == 99% of the userbase), so what does it offer that they will find interesting? The fact that it defines new road-types, I think. Which brings back the topic above, give better insight of the train/roadveh types are introduced in a set. Need to think about it, but it is a solid point. Just the more difficult one :D

As for NRS, we simply haven't add the classifier for that yet :D Will fix that soon, so it becomes properly classified .. `unknown` can be considered a bug for that set atm :)

**Reply by EratoNysiad** [reply]
The dropdown list is a very good one. For example, to make my point for NRT, I typed in "road" and noticed how that doesn't really work, I got a lot of things related to road-vehicles. The "road-infra" tag is good, I know how the system works, and how if I suffixed "-infra" to it, it would work just fine. The average user, probably not.

I do agree that indeed most players won't understand what NRT means. Simultaneously I will argue that more experienced players may not know the name, but will definitely know the phenomenon. I think it's incorrect to say that the "Trolleybus Set" does not add trolleybuses, but I would say it does not do what the majority of people looking for trolleybuses are looking for. A naive classification system would put them under "trams", which I would say is also wrong. I would also say it's wrong to say the newgrf is necessarily bad.
I think this goes for pretty much everything that NRT offers; pre-NRT newgrfs are often just a worse player experience, and I believe the in-game content downloader can help in ensuring a good player experience.

**Reply by TrueBrain** [reply]
What would you suggest would help the end-user the most, in case of NRT?

**Reply by EratoNysiad** [reply]
I'm not sure, maybe a `uses tram-types` and `uses road-types` tag, which could be used for both infrastructure and vehicle sets?

**Reply by frosch123** [reply]
Technically the auto-tagger can detect whether a vehicle set and/or road-infra set uses NRT.

I'll add two half-baked ideas to the mix:
* I can see that someone who knows or reads about specific features, would want to explicitly search for them, just to try them out. As such we could expose the low-level categories like "NRT", "NewHouses", "NRS" in some "expert" category. So users can either generally search for "set-type:road-infra" or for "newgrf-feature:RoadTypes".
* There some old NewGRF which do things the old way: Replacing houses, roads, rails, stations, ...  by replacing base-sprites. And there are modern variants for these: NewHouses, NewStations, NewRoadTypes. We could label the old ones "classic": they are still nice to use in their original context, but they do not work as well with other features like the new versions do. So we add have "classic-infra" which accumulates NewGRF replacing stations, road/rail, depot, ... sprites by replacing base-sprites.

**Reply by TrueBrain** [reply]
https://bananas.openttd.org/package/newgrf?set=road-stop is now live, fixing the mentioned misclassified NRS :)

**Reply by Tintinfan** [reply]
I know most of the points were covered by Erato, but I thought I'd add to the pool of thought and help improve on your work. I also do like the classification system, particulary how it standardises what groups GRFs are in, their extrazoom and bpp - without the user having to define it and without there being a large variance of differing words used in tags leading to some long searches for things. It has hard to adapt to change, espically when at first it doesn't do what you expect anymore. Regardless, lets see if we can make this better than it ever was before. Good work. :) 

I don't think we need to return to a tagging world, espically if you can sort out reading from the NewGRF description and allowing that to be part of the search terms. As already discussed, geotagging by country was perhaps the main reason for the "lets go back to tags, this is awful!" sentiment - as being able to search for sets by geography was always fun. I think that being able to tag multiple countries will be important for implementing this. I also thought geographic regions might be good, but there are not many sets covering a whole continent - and if this is included in the GRF description it could be searched for anyway. There might also be some other sub-categories people might like that tags helped with; such as "narrow gauge" or "electric trains" - but I think these are less important. and would need more discussion.

![image](https://user-images.githubusercontent.com/1836032/222885218-d161a7c2-9425-4661-9ec1-f0a58c250196.png)
This panel on the online version is very much was changed my mind - it makes it very clear what you are searching for and would be very benefical to have in-game. I also believe part of the issue was not quite knowing what the search terms for things are now in-game; and having some way to filter using these options in a drop-down like on the website would make the search bar in the content manager a suppliment to finding the NewGRF you want as opposed to the main way. An example of where this is an issue would be search for tracksets. When searching for "track" in the search box, you assume you'll get all sets under "rail infra" - including popular GRFs like U&ReRMM, which is correctly tagged as "rail infra" but nowhere does it have "track" in the name, so it doesn't appear as a result in search. It's not obvious that "rail infra" is what you should be search for to get all listing of tracksets.


TLDR;
- Search box should read NewGRF description
- Country of NewGRF should be a searchable category
- Other categories that could be nice to have
- Categories are not clear; could these be listed in-game? or perhaps some way to translate "track" => "rail infra" for example.

**Reply by TrueBrain** [reply]
Some progress report: regions are mostly implemented; needs some testing, but seems to be working fine.

I converted a lot of the old tags that indicated regions to the new region system, to help the content listing along. There might be some misses in there, but I think I got most right.

![image](https://user-images.githubusercontent.com/1663690/222929571-46a165fa-9a64-4c85-a723-44a8aa9272d0.png)

That is how it would look in-game, returning the ability to search for "Japan" and get a lot more hits :)

**Reply by TrueBrain** [reply]
This change is now live. We will continue to work on the rest, but as this was the most important one, we prioritized it :)
